system-addon/lib/TopSitesFeed.jsm

```diff
@@ -55,7 +61,12 @@ this.TopSitesFeed = class TopSitesFeed {
       frecent = frecent.filter(link => link && link.type !== "affiliate");
     }
 
-    return insertPinned([...frecent, ...DEFAULT_TOP_SITES], pinned).slice(0, TOP_SITES_SHOWMORE_LENGTH);
+    pinned = insertPinned([...frecent, ...DEFAULT_TOP_SITES], pinned);
+
+    // Parse site url and extract hostname.
+    pinned.forEach(site => { site.hostname = shortURL(site); });
+
+    return this.dedupe.collection(pinned).slice(0, TOP_SITES_SHOWMORE_LENGTH);
```

Comment:
Currently no, but Dedupe can be called with a compare function that prefers pinned items.